graspologic/plot/plot_matrix.py

```diff
@@ -0,0 +1,933 @@
+
+    item = list(item_meta.columns.values)
+
+    return item_meta, item
+
+
+def _set_spines(ax, left, right, top, bottom):
+    ax.spines["right"].set_visible(right)
+    ax.spines["top"].set_visible(top)
+    ax.spines["left"].set_visible(left)
+    ax.spines["bottom"].set_visible(bottom)
+
+    return ax
+
+
+def _remove_shared_ax(ax):
+    """
+    Remove ax from its sharex and sharey
+    """
+    # Remove ax from the Grouper object
+    shax = ax.get_shared_x_axes()
+    shay = ax.get_shared_y_axes()
+    shax.remove(ax)
+    shay.remove(ax)
+
+    # Set a new ticker with the respective new locator and formatter
+    for axis in [ax.xaxis, ax.yaxis]:
+        ticker = mpl.axis.Ticker()
+        axis.major = ticker
+        axis.minor = ticker
+        # No ticks and no labels
+        loc = mpl.ticker.NullLocator()
+        fmt = mpl.ticker.NullFormatter()
+        axis.set_major_locator(loc)
+        axis.set_major_formatter(fmt)
+        axis.set_minor_locator(loc)
+        axis.set_minor_formatter(fmt)
+
+
+def draw_colors(
+    ax, ax_type="x", meta=None, divider=None, color_class=None, palette="tab10"
+):
+    """
+    Draw colormap onto the axis to separate the data
+
+    Parameters
+    ----------
+    ax : matplotlib axes object
+        Axes in which to draw the colormap
+    ax_type : char, optional
+        Setting either the x or y axis, by default "x"
+    meta : pd.DataFrame, pd.Series, list of pd.Series or np.array, optional
+        Metadata of the matrix such as class, cell type, etc., by default None
+    divider : AxesLocator, optional
+        Divider used to add new axes to the plot
+    color_class : string or list of string, optional
+        Attribute in meta by which to draw colorbars, by default None
+    palette : str, optional
+        Colormap of the colorbar, by default "tab10"
+
+    Returns
+    -------
+    ax : matplotlib axes object
+        Axes in which to draw the color map
+    """
+    classes = meta[color_class]
+    uni_classes = np.unique(classes)
+    # Create the color dictionary
+    if isinstance(palette, dict):
+        color_dict = palette
+    elif isinstance(palette, str):
+        color_dict = dict(
+            zip(uni_classes, sns.color_palette(palette, len(uni_classes)))
+        )
+
+    # Make the colormap
+    class_map = dict(zip(uni_classes, range(len(uni_classes))))
+    color_sorted = np.vectorize(color_dict.get)(uni_classes)
+    color_sorted = np.array(color_sorted)
+    if len(color_sorted) != len(uni_classes):
+        color_sorted = color_sorted.T
+    lc = ListedColormap(color_sorted)
+    class_indicator = np.vectorize(class_map.get)(classes)
+
+    if ax_type == "x":
+        class_indicator = class_indicator.reshape(1, len(classes))
+    elif ax_type == "y":
+        class_indicator = class_indicator.reshape(len(classes), 1)
+    sns.heatmap(
+        class_indicator,
+        cmap=lc,
+        cbar=False,
```

Comment:
Do you think it would be helpful to include a cbar for the classes sometimes?

Reply:
It would be helpful, but the cbar sometimes makes the figure only shows up partially.

Reply:
Where were you trying to put the cbar?

Reply:
I didn't regulate the position of the cbar, and it would usually show up on the right side. I usually see partial images in VS Code, so it could be a problem with VS Code rather than the program itself.

Reply:
i see the utility of this, but I think we should open this as a separate issue. would need to figure out how to deal with many different possible color maps if many different colors are present. also, setting `cbar` to True would look bad, we'll need to do it ourselves on a custom axis.